sources/bloodhound/src/bin/bottlerocket-checks/checks.rs

```diff
@@ -0,0 +1,1280 @@
+        {
+            let mut first_found = false;
+            let mut second_found = false;
+
+            if output.status.success() {
+                let std_output = String::from_utf8_lossy(&output.stdout).to_string();
+                for line in std_output.lines() {
+                    if !first_found && line.contains(first.0) && line.contains(first.1) {
+                        first_found = true;
+                        continue;
+                    }
+
+                    if first_found && line.contains(second.0) && line.contains(second.1) {
+                        second_found = true;
+                        break;
+                    }
+                }
+            }
+
+            if first_found && second_found {
+                result.status = results::CheckStatus::PASS;
+            } else {
+                result.error = "Unable to find expected iptables INPUT values".to_string();
+                result.status = results::CheckStatus::FAIL;
+                return result;
+            }
+        } else {
+            result.error = "unable to verify iptables INPUT settings".to_string();
+        }
+
+        if let Some(found) = look_for_string_in_output(
+            IP6TABLES_CMD,
+            ["-L", "OUTPUT", "-v", "-n"],
+            "ACCEPT     0    --  *      lo      ::/0                 ::/0",
+        ) {
+            if !found {
+                result.error = "iptables OUTPUT rule not found".to_string();
+                result.status = results::CheckStatus::FAIL;
+            } else {
+                result.status = results::CheckStatus::PASS;
+            }
+        } else {
+            result.error =
+                "unable to parse iptables OUTPUT rules to verify loopback policy".to_string();
+        }
+
+        result
+    }
+
+    fn metadata(&self) -> results::CheckerMetadata {
+        results::CheckerMetadata {
+            title: "Ensure IPv6 loopback traffic is configured".to_string(),
+            id: "3.4.2.2".to_string(),
+            level: 2,
+            name: "br03040202".to_string(),
+            mode: results::Mode::Automatic,
+        }
+    }
+}
+
+// =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<=
+
+pub struct BR04010101Checker {}
+
+impl results::Checker for BR04010101Checker {
+    fn execute(&self) -> results::CheckerResult {
+        let mut result = results::CheckerResult {
+            error: String::new(),
+            status: results::CheckStatus::SKIP,
+        };
+
+        if let Some(found) = look_for_string_in_file(JOURNALD_CONF_FILE, "Storage=persistent") {
+            if !found {
+                result.error = "journald is not configured".to_string();
+                result.status = results::CheckStatus::FAIL;
+            } else {
+                result.status = results::CheckStatus::PASS;
+            }
+        } else {
+            result.error = "unable to verify journald settings".to_string();
+        }
```

Comment:
This same sort of pattern is repeated so often that it might be worth turning into a macro.
```
check_file_contains!(path, strings_to_match, unable_to_check_error, did_not_find_error)

check_file_contains!(JOURNALD_CONF_FILE, "Storage=persistent", "unable to verify journald settings", "journald is not configured")
```

Then most of these checks would read like asserts in unit tests.